### PR TITLE
Fixes #15478: missing documentation about api settings 

### PR DESCRIPTION
**Pull request by ncharles**
https://issues.rudder.io/issues/15478

### CODE REVIEW

File: src/api_v2/settings.md [hunk 1]
```diff
@@ -104,6 +108,21 @@ https://github.com/Normation/rudder/blob/branches/rudder/4.1/rudder-web/src/main
...
   @apiParam {String} [send_metrics] Send anonymous usage statistics
 
+  @apiParam {String} [node_onaccept_default_state] Set default state for node when they are accepted within Rudder, can be "enabled", "ignored", "empty-policies", "initializing", "preparing-eol"
+  @apiParam {String} [node_onaccept_default_policyMode] Default policy mode for accepted node, can be "default", "enforce" or "audit"
+
+  @apiParam {Boolean} [unexpected_allows_duplicate] Ignore duplicated compliance report (default true)
+  @apiParam {Boolean} [unexpected_unbound_var_values] Allows multiple reports for configuration based on a multivalued variable: (default true)
+
+  @apiParam {Boolean} [rudder_compute_changes] Compute list of changes (repaired reports) per rules (default true)
+  @apiParam {Boolean} [rudder_generation_compute_dyngroups] Recompute all dynamic gorups at the start of policy generation (default true)
```
Comment (amousset) [inline src/api_v2/settings.md]: groups

File: src/api_v2/settings.md [hunk 2]
```diff
@@ -152,6 +176,20 @@ https://github.com/Normation/rudder/blob/branches/rudder/4.1/rudder-web/src/main
...
+  @apiParam {Boolean} [rudder_save_db_compliance_levels] Store in database all compliance levels (default true)
```
Comment (amousset) [inline src/api_v2/settings.md]: Store all compliance levels in database

File: src/api_v2/settings.md [hunk 3]
```diff
@@ -152,6 +176,20 @@ https://github.com/Normation/rudder/blob/branches/rudder/4.1/rudder-web/src/main
...
+  @apiParam {Boolean} [rudder_save_db_compliance_details] Store in database all compliance details (default false)
```
Comment (amousset) [inline src/api_v2/settings.md]: same

File: src/api_v2/settings.md [hunk 4]
```diff
@@ -152,6 +176,20 @@ https://github.com/Normation/rudder/blob/branches/rudder/4.1/rudder-web/src/main
   @apiParam {String} [enable_javascript_directives] Enable script evaluation in Directives
   @apiParam {String} [send_metrics] Send anonymous usage statistics
 
+  @apiParam {String} [node_onaccept_default_state] Set default state for node when they are accepted within Rudder, can be "enabled", "ignored", "empty-policies", "initializing", "preparing-eol"
+  @apiParam {String} [node_onaccept_default_policyMode] Default policy mode for accepted node, can be "default", "enforce" or "audit"
+
+  @apiParam {Boolean} [unexpected_allows_duplicate] Ignore duplicated compliance report (default true)
```
Comment (amousset) [inline src/api_v2/settings.md]: reports

File: src/api_v2/settings.md [hunk 5]
```diff
@@ -152,6 +176,20 @@ https://github.com/Normation/rudder/blob/branches/rudder/4.1/rudder-web/src/main
...
+  @apiParam {Boolean} [unexpected_unbound_var_values] Allows multiple reports for configuration based on a multivalued variable: (default true)
```
Comment (amousset) [inline src/api_v2/settings.md]: s/://

File: src/api_v2/settings.md [hunk 6]
```diff
@@ -152,6 +176,20 @@ https://github.com/Normation/rudder/blob/branches/rudder/4.1/rudder-web/src/main
...
+
+  @apiParam {Boolean} [rudder_compute_changes] Compute list of changes (repaired reports) per rules (default true)
+  @apiParam {Boolean} [rudder_generation_compute_dyngroups] Recompute all dynamic gorups at the start of policy generation (default true)
```
Comment (amousset) [inline src/api_v2/settings.md]: gorups/groups

File: src/api_v2/settings.md [hunk 7]
```diff
@@ -152,6 +176,20 @@ https://github.com/Normation/rudder/blob/branches/rudder/4.1/rudder-web/src/main
...
+  @apiParam {Boolean} [rudder_save_db_compliance_levels] Store in database all compliance levels (default true)
+  @apiParam {Boolean} [rudder_save_db_compliance_details] Store in database all compliance details (default false)
+
+  @apiParam {String} [rudder_generation_max_parallelism] Set the policy generation parrallelism, either in as an number of thread (i.e. 4), or a multiplicative of the number of core (x0.5) (default 0.5)
+  @apiParam {Int} [rudder_generation_js_timeout] Policy generation JS evaluation of directive parameter timeout in seconds(default 30)
```
Comment (amousset) [inline src/api_v2/settings.md]: missing space before parenthesis

File: src/api_v2/settings.md [hunk 8]
```diff
@@ -152,6 +176,20 @@ https://github.com/Normation/rudder/blob/branches/rudder/4.1/rudder-web/src/main
...
+  @apiParam {Boolean} [rudder_generation_continue_on_error] Policy generation continues on NodeConfiguration evaluation (default false)
```
Comment (amousset) [inline src/api_v2/settings.md]: "continues on error"? (or am I missing something?)

File: src/api_v2/settings.md [hunk 9]
```diff
@@ -152,6 +176,20 @@ https://github.com/Normation/rudder/blob/branches/rudder/4.1/rudder-web/src/main
...
+
+  @apiParam {Boolean} [unexpected_allows_duplicate] Ignore duplicated compliance report (default true)
+  @apiParam {Boolean} [unexpected_unbound_var_values] Allows multiple reports for configuration based on a multivalued variable: (default true)
+
+  @apiParam {Boolean} [rudder_compute_changes] Compute list of changes (repaired reports) per rules (default true)
+  @apiParam {Boolean} [rudder_generation_compute_dyngroups] Recompute all dynamic gorups at the start of policy generation (default true)
+  @apiParam {Boolean} [rudder_save_db_compliance_levels] Store in database all compliance levels (default true)
+  @apiParam {Boolean} [rudder_save_db_compliance_details] Store in database all compliance details (default false)
+
+  @apiParam {String} [rudder_generation_max_parallelism] Set the policy generation parrallelism, either in as an number of thread (i.e. 4), or a multiplicative of the number of core (x0.5) (default 0.5)
```
Comment (amousset) [inline src/api_v2/settings.md]: s/in//

File: src/api_v2/settings.md [hunk 10]
```diff
@@ -128,9 +147,14 @@ https://github.com/Normation/rudder/blob/branches/rudder/4.1/rudder-web/src/main
 
   @apiParam {Boolean} [require_time_synchronization] Require time synchronization between nodes and policy server
 
+  @apiParam {String} [relay_server_synchronization_method] Method used to synchronize data between Rudder Server and Relays, either "classic" (agent protocol, default), "rsync" (use rsync to synchronize, that you'll need to be manually set up), or "disabled" (use third party system to transmit data)
+  @apiParam {Boolean} [relay_server_synchronize_policies] If "rsync" is set as a synchronization method, use rsync to synchronize policies between Rudder Server and Relays. If false, you'll have to synchronize yourself policies.
```
Comment (amousset) [inline src/api_v2/settings.md]: synchronize policies yourself

File: src/api_v2/settings.md [hunk 11]
```diff
@@ -128,9 +147,14 @@ https://github.com/Normation/rudder/blob/branches/rudder/4.1/rudder-web/src/main
...
+  @apiParam {Boolean} [relay_server_synchronize_shared_files] If "rsync" is set as a synchronization method, use rsync to synchronize shared files between Rudder Server and Relays. If false, you'll have to synchronize yourself policies.
```
Comment (amousset) [inline src/api_v2/settings.md]: same

File: src/api_v2/settings.md (hunk 8 above)
Comment (ncharles) [reply src/api_v2/settings.md]: indeed on error

**Comment by ncharles** [comment]
Commit modified

**Pull request merged**